### PR TITLE
Fixes sharing text feature

### LINKED ISSUE #934: Feature Sharing doesn't work
### Details

Sharing doesn't work on the text features. the `ShareButton` accepts a piece of content and then goes and pulls the data it needs from the title. And the message is is just `title + url`. the `message` prop isn't even being used.

### Expected Behavior

When you share a feature, it should share the content of the feature. 

### Actual Behavior

It shares the content title.

Kapture 2019-07-29 at 15 40 20

### Possible Fix

<details><summary>Additional Info</summary>

### Your Environment

- Environment name and version (e.g. Chrome 39, node.js 5.4):
- Operating System and version (desktop or mobile):
- Link to your project:

### Steps to Reproduce

1.  first...
2.
3.
4.

### Stack Trace

</details>

### PR DESCRIPTION
**Pull request by vinnyjth**
### What does this PR do, or why is it needed?

Updates the `share` function so it accepts a `message` property and fixes a typo that was preventing `message` from being passed down. 

### What design trade-offs/decisions were made?

Chose to now engage in a more full featured refactor like @redreceipt did in #935. Trying to find a way to get the intended functionality working without a full scale refactor. 

### How do I test this PR?

1. Open a content item with a text feature
2. Click the share button
3. click "copy to clipboard" 
3. paste somewhere else on your ios device (see screenshot for details) 

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] Closes #934
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by redreceipt** [comment]
in thinking about this, don't you think we should not do anything fancy with the `Share.share` function and handle all the platform stuff at the source? so:

`Platform.OS === android ? share({title, message: message+url}) : share({title, message, url})`

**Comment by redreceipt** [comment]
wait nevermind, I see the value of it, but I have some suggestions...

File: packages/apolloschurchapp/src/utils/content/share.js [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 
-const share = ({ title, url }) => {
+const share = ({ title, url, message }) => {
   Share.share({
     title,
-    message: Platform.OS === 'android' ? `${title}\n${url}` : title,
+    message:
+      Platform.OS === 'android'
+        ? `${message || title}\n${url}`
+        : message || title,
```
Comment (redreceipt) [inline packages/apolloschurchapp/src/utils/content/share.js]:
```suggestion
        : message,
```

File: packages/apolloschurchapp/src/utils/content/share.js [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
-// import getSiteLink from './getSiteLink';
 
-const share = ({ title, url }) => {
+const share = ({ title, url, message }) => {
   Share.share({
     title,
-    message: Platform.OS === 'android' ? `${title}\n${url}` : title,
+    message:
+      Platform.OS === 'android'
+        ? `${message || title}\n${url}`
```
Comment (redreceipt) [inline packages/apolloschurchapp/src/utils/content/share.js]:
```suggestion
        ? `${message}\n${url}`
```

File: packages/apolloschurchapp/src/utils/content/share.js [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-// import { track, events } from 'apolloschurchapp/src/analytics';
-// import getSiteLink from './getSiteLink';
 
-const share = ({ title, url }) => {
+const share = ({ title, url, message }) => {
   Share.share({
     title,
-    message: Platform.OS === 'android' ? `${title}\n${url}` : title,
+    message:
+      Platform.OS === 'android'
```
Comment (redreceipt) [inline packages/apolloschurchapp/src/utils/content/share.js]:
```suggestion
      Platform.OS === 'android' && url
```

**Comment by redreceipt** [comment]
I added some comments that will break content sharing as it stands but the fix should be simple, we should be passing `message` to the share function, not `title`. We are intending it to be the message anyway.

**Comment by vinnyjth** [comment]
@michael-riddering I added a `message` to the ContentItem sharing resolver.

**Comment by redreceipt** [comment]
love it, maybe I'm splitting hairs but I still think our implementation is a little confusing. 

the API field should be called `summary`, not `message`
https://github.com/ApollosProject/apollos-prototype/blob/9aa3c6abd4d8634d9916068f5a414c4fba7b4787/packages/apollos-data-schema/src/index.js#L403

the `ContentItem` should be passing `title` as `message` from the API into the `content` prop of `<Share>, it it's odd behavior to share the content and only paste the first sentence of the article. so we are explicit about the fact that we are using the title as the shared message body which I think is a better UX.
https://github.com/ApollosProject/apollos-prototype/blob/9aa3c6abd4d8634d9916068f5a414c4fba7b4787/packages/apolloschurchapp/src/ui/ShareButton/index.js#L18

**Comment by redreceipt** [comment]
if you want, I can make those suggestions apart of a refactor PR, I've already started that other one and I still think we should rename `<Share>` and `<ShareButton>`

**Comment by vinnyjth** [comment]
> the API field should be called summary, not message
The `sharing` interface is meant to be used more broadly than ContentItem, hence why we call that field `message`.

>the ContentItem should be passing title as message from the API into the content prop of ShareButton

~I don't think I agree with this. I think we should pass the `summary` as the message, and the content `title` as the title.~ I misunderstand the title prop. You're right about needing the `title` of the content item in the `message` field.

**Comment by vinnyjth** [comment]
@redreceipt I this this is good now!
